src/main/java/seedu/duke/commands/ResetCommand.java

```diff
@@ -0,0 +1,49 @@
+package seedu.duke.commands;
+
+import seedu.duke.exceptions.ModHappyException;
+import seedu.duke.tasks.Module;
+import seedu.duke.tasks.ModuleList;
+import seedu.duke.tasks.TaskList;
+import seedu.duke.util.StringConstants;
+
+import java.util.ArrayList;
+
+public class ResetCommand extends Command {
+    @Override
+    public CommandResult execute(ModuleList moduleList) throws ModHappyException {
+        removeAll(moduleList);
+        return new CommandResult(StringConstants.RESET_MESSAGE);
+    }
+
+    /**
+     * Remove all tasks and modules from the list.
+     *
+     * @param moduleList List from which modules and tasks are to be deleted from.
+     */
+    public void removeAll(ModuleList moduleList) throws ModHappyException {
+        removeGeneralTasks(moduleList);
+        ArrayList<Module> modules = moduleList.getModuleList();
+        for (Module m: modules) {
+            TaskList moduleTaskList = m.getTaskList();
+            for (int i = 0; i < moduleTaskList.size(); i += 1) {
+                moduleTaskList.removeTask(i);
+            }
+        }
```

Comment:
I remember there is no need to do the nested clear for java, because java has JGC, any un-referenced object will be recycled(e.g. here each modules and taskList in the modules). So moduleList.clear() will achieve the same effort as the nested loop?

Reply:
Yup, just using moduleList.clear() here works, thanks for point it out.